vowpalwabbit/json_parser/tests/dsjson_parser_test.cc

```diff
@@ -69,59 +71,59 @@ BOOST_AUTO_TEST_CASE(parse_dsjson_p_duplicates)
 }
   )";
   auto vw = VW::initialize("--dsjson --chain_hash --cb_adf --no_stdin --quiet", nullptr, false, nullptr, nullptr);
-  VW::details::decision_service_interaction interaction;
+  VW::parsers::json::decision_service_interaction interaction;
 
-  auto examples = parse_dsjson(*vw, json_text, &interaction);
+  auto examples = vwtest::parse_dsjson(*vw, json_text, &interaction);
   VW::finish_example(*vw, examples);
   VW::finish(*vw);
 
   // Use the latest "p" or "_p" field provided. The "_p" is ignored when it's inside "c".
   static constexpr float EXPECTED_PDF[2] = {0.5f, 0.5f};
   const size_t num_probabilities = interaction.probabilities.size();
-  BOOST_CHECK_EQUAL(num_probabilities, 2);
+  EXPECT_EQ(num_probabilities, 2);
   for (size_t i = 0; i < num_probabilities; ++i)
   {
     // Check that probabilities are as expected.
-    BOOST_CHECK_EQUAL(interaction.probabilities[i], EXPECTED_PDF[i]);
+    EXPECT_EQ(interaction.probabilities[i], EXPECTED_PDF[i]);
   }
 }
 
-BOOST_AUTO_TEST_CASE(parse_dsjson_pdrop_float)
+TEST(dsjson_tests, parse_dsjson_pdrop_float)
 {
   const std::string json_text = R"(
 {
   "pdrop": 0.1
 }
   )";
   auto vw = VW::initialize("--dsjson --chain_hash --cb_adf --no_stdin --quiet", nullptr, false, nullptr, nullptr);
-  VW::details::decision_service_interaction interaction;
+  VW::parsers::json::decision_service_interaction interaction;
 
-  auto examples = parse_dsjson(*vw, json_text, &interaction);
+  auto examples = vwtest::parse_dsjson(*vw, json_text, &interaction);
   VW::finish_example(*vw, examples);
   VW::finish(*vw);
 
-  BOOST_CHECK_CLOSE(0.1f, interaction.probability_of_drop, FLOAT_TOL);
+  EXPECT_NEAR(0.1f, interaction.probability_of_drop, vwtest::FLOAT_TOL);
 }
 
-BOOST_AUTO_TEST_CASE(parse_dsjson_pdrop_uint)
+TEST(dsjson_tests, parse_dsjson_pdrop_uint)
 {
   const std::string json_text = R"(
 {
   "pdrop": 0
 }
   )";
   auto vw = VW::initialize("--dsjson --chain_hash --cb_adf --no_stdin --quiet", nullptr, false, nullptr, nullptr);
-  VW::details::decision_service_interaction interaction;
+  VW::parsers::json::decision_service_interaction interaction;
 
-  auto examples = parse_dsjson(*vw, json_text, &interaction);
+  auto examples = vwtest::parse_dsjson(*vw, json_text, &interaction);
   VW::finish_example(*vw, examples);
   VW::finish(*vw);
 
-  BOOST_CHECK_CLOSE(0.0f, interaction.probability_of_drop, FLOAT_TOL);
+  EXPECT_NEAR(0.0f, interaction.probability_of_drop, vwtest::FLOAT_TOL);

```

Comment:
Replace with EXPECT_FLOAT_EQ
```suggestion
  EXPECT_FLOAT_EQ(0.0f, interaction.probability_of_drop);

```